fhir-search/src/test/java/com/ibm/fhir/search/test/InclusionParameterParseTest.java

```diff
@@ -848,4 +850,412 @@ public void testWildcardIncludeAndRevIncludeNoTargetType() throws Exception {
+        String validQueryString3 = "&_include:iterate=Patient:link:Patient";
+        String validQueryString4 = "&_include:iterate=Patient:link:RelatedPerson";
+
+        List<InclusionParameter> expectedIncludeParms = new ArrayList<>();
+        expectedIncludeParms.add(new InclusionParameter("Organization", "endpoint", "Endpoint", Modifier.ITERATE, false));
+        expectedIncludeParms.add(new InclusionParameter("Patient", "organization", "Organization", null, false));
+        expectedIncludeParms.add(new InclusionParameter("Patient", "link", "Patient", Modifier.ITERATE, false));
+        expectedIncludeParms.add(new InclusionParameter("Patient", "link", "RelatedPerson", Modifier.ITERATE, false));
+
+        queryParameters.put("_include:iterate", Arrays.asList(new String[] { "Organization:endpoint", "Patient:link" }));
+        queryParameters.put("_include", Collections.singletonList("Patient:organization"));
+        searchContext = SearchUtil.parseQueryParameters(resourceType, queryParameters);
+        assertNotNull(searchContext);
+        assertTrue(searchContext.hasIncludeParameters());
+        assertFalse(searchContext.hasRevIncludeParameters());
+        assertEquals(expectedIncludeParms.size(), searchContext.getIncludeParameters().size());
+        for (InclusionParameter incParm : searchContext.getIncludeParameters()) {
+            assertTrue(expectedIncludeParms.contains(incParm));
+        }
+
+        String selfUri = SearchUtil.buildSearchSelfUri("http://example.com/Patient", searchContext);
+        assertTrue(selfUri.contains(validQueryString1));
+        assertTrue(selfUri.contains(validQueryString2));
+        assertTrue(selfUri.contains(validQueryString3));
+        assertTrue(selfUri.contains(validQueryString4));
+    }
+
+    @Test
+    public void testIterateIncludeMissingTargetType() throws Exception {
+        Map<String, List<String>> queryParameters = new HashMap<>();
+        Class<Patient> resourceType = Patient.class;
+        FHIRSearchContext searchContext;
+        String validQueryString1 = "&_include:iterate=Patient:general-practitioner:Organization&";
+        String validQueryString2 = "&_include:iterate=Patient:general-practitioner:Practitioner&";
+        String validQueryString3 = "&_include:iterate=Patient:general-practitioner:PractitionerRole&";
+
+        List<InclusionParameter> expectedIncludeParms = new ArrayList<>();
+        expectedIncludeParms.add(new InclusionParameter("Patient", "general-practitioner", "Organization", Modifier.ITERATE, false));
+        expectedIncludeParms.add(new InclusionParameter("Patient", "general-practitioner", "Practitioner", Modifier.ITERATE, false));
+        expectedIncludeParms.add(new InclusionParameter("Patient", "general-practitioner", "PractitionerRole", Modifier.ITERATE, false));
+
+        queryParameters.put("_include:iterate", Collections.singletonList("Patient:general-practitioner"));
```

Comment:
Do we have any tests involving possible circular references? Although it's probably not an issue if we only iterate one level. Am I right in thinking that your unique ids check would block any possible infinite loop anyway?

Reply:
If we decide to raise the max iterations to more than 1 I can add a testcase. And, yes, it should already be handled by current design.